src/ArrayComponents.ml

```diff
@@ -0,0 +1,32 @@
+open Base
+
+open Expr
+
+let all = [
+  {
+    name = "select";
+    codomain = Type.TVAR ("b");
+    domain = [Type.ARRAY (Type.TVAR "a", Type.TVAR "b"); Type.TVAR "a"];
+    is_argument_valid = (function
+                         | _ -> true);
+    evaluate = (function [@warning "-8"]
+                [Value.Array (_, _, elems, default_val) ; key]
+                -> match List.Assoc.find elems ~equal:Value.equal key with
+                   | None -> default_val
+                   | Some value -> value);
+    to_string = (fun [@warning "-8"] [a ; b] -> "(select " ^ a ^ " " ^ b ^ ")");
+    global_constraints = (fun _ -> [])
+  }
+  ;
+  {
+    name = "store";
+    codomain = Type.ARRAY (Type.TVAR "'a", Type.TVAR "'b");
+    domain = [Type.ARRAY (Type.TVAR "'a", Type.TVAR "'b"); Type.TVAR "'a"; Type.TVAR "'b"];
```

Comment:
We should be consistent with naming TVARs. The `select` component doesn't use apostrophe(') but `store` does. You should remove those.

Reply:
Okay you added more apostrophes instead of removing from `select`, but that's fine ... as long as we are consistent :slightly_smiling_face: